relayer/src/services/relayer/optimism.ts

```diff
@@ -76,6 +76,32 @@ export class OptimismRelayerService extends BaseRelayerService {
     return;
   }
 
+  async safeRelayMessage(message: CrossChainMessage): Promise<boolean> {
+    try {
+      // Finalize message via OptimismPortal (it calls the L1CrossDomainMessenger internally)
+      (await this.messenger.finalizeMessage(message)).wait(1);
```

Comment:
I assume you could also check it by reading `portal.finalizedWithdraws[hash]`

Reply:
The problem with this is that i will also need to do another check to see if it was relayed so we know if we actually need to call the OP CrossDomainMessenger.
Also, how do i get the hash?

Reply:
Found a utility function for hashing. Lets see about the other parts.

Reply:
So, for the first part:
>The problem with this is that i will also need to do another check to see if it was relayed so we know if we actually need to call the OP CrossDomainMessenger

I figured i dont need to do this check as i am already checking `if (status === MessageStatus.RELAYED)` before calling this function which means we will never execute this function if the message was already relayed.

---

For the second part:
> I am trying to see if there are any utility functions we can use from the SDK but i cannot find any fitting solution.

I found a way to hash the low level message in a nice and easy to understand way. Checking if there was a withdrawal is done directly via the OptimismPortal contract by checking against `finalizedWithdrawals[hash]`. If this is set true, then i relay via the Messenger.

Reply:
Done. Please take a look again and read my last comments on this PR.